src/lib.rs

```diff
@@ -504,31 +530,39 @@ impl Member {
                 bits,
                 base_ty,
                 default,
+                access,
                 inner: None,
             })
         }
     }
 
     fn debug(&self) -> TokenStream {
-        if let Some(inner) = &self.inner {
-            let ident_str = inner.ident.to_string();
-            let ident = &inner.ident;
-            quote!(.field(#ident_str, &self.#ident()))
-        } else {
-            quote!()
+        match &self.inner {
+            Some(inner) if matches!(self.access, AccessMode::ReadWrite | AccessMode::ReadOnly) => {
```

Comment:
If you already have a match here, you could just match over both inner and access like so:

```rs
match (&self.inner, self.access) {
    (Some(inner), AccessMode::ReadWrite | AccessMode::ReadOnly) => { /*...*/ }
    _ => quote!(),
}
```